bufferflow_timedraw.go

```diff
@@ -8,40 +8,55 @@ import (
 )
 
 type BufferflowTimedRaw struct {
-	Name   string
-	Port   string
-	Output chan []byte
-	Input  chan string
-	ticker *time.Ticker
+	port              string
+	output            chan []byte
+	input             chan string
+	done              chan bool
+	ticker            *time.Ticker
+	bufferedOutputRaw []byte
+	sPortRaw          string
 }
 
-var (
-	bufferedOutputRaw []byte
-)
+func NewBufferflowTimedRaw(port string, output chan []byte) *BufferflowTimedRaw {
+	return &BufferflowTimedRaw{
+		port:              port,
+		output:            output,
+		input:             make(chan string),
+		done:              make(chan bool),
+		ticker:            time.NewTicker(16 * time.Millisecond),
+		bufferedOutputRaw: nil,
+		sPortRaw:          "",
+	}
+}
 
 func (b *BufferflowTimedRaw) Init() {
-	log.Println("Initting timed buffer flow (output once every 16ms)")
-
-	go func() {
-		for data := range b.Input {
-			bufferedOutputRaw = append(bufferedOutputRaw, []byte(data)...)
-		}
-	}()
-
-	go func() {
-		b.ticker = time.NewTicker(16 * time.Millisecond)
-		for _ = range b.ticker.C {
-			if len(bufferedOutputRaw) != 0 {
-				m := SpPortMessageRaw{b.Port, bufferedOutputRaw}
+	log.Println("Initting timed buffer raw flow (output once every 16ms)")
+	go b.consumeInput()
+}
+
+func (b *BufferflowTimedRaw) consumeInput() {
+Loop:
```

Comment:
This `Loop:` must be after the below `for` otherwise you'll be stuck in this function forever even after the `break Loop` at line 41. Same for other `Buffer`s.

Reply:
https://www.ardanlabs.com/blog/2013/11/label-breaks-in-go.html